src/binary-reader-ir.cc

```diff
@@ -660,6 +698,10 @@ Result BinaryReaderIR::OnNopExpr() {
   return AppendExpr(expr);
 }
 
+Result BinaryReaderIR::OnRethrowExpr(Index depth) {
+  return AppendExpr(new RethrowExpr(Var(depth)));
```

Comment:
You can construct `Var(depth, GetLocation())`. I noticed that I forgot to do this for other `Var` constructors in this file too.